app/controllers/books_controller.rb

```diff
@@ -2,4 +2,13 @@ class BooksController < ApplicationController
   def index
     @pagy, @books = pagy Book.order_name, items: Settings.length.digit_12
   end
+
+  def show
+    @book = Book.find_by id: params[:id]
+    @books = Book.group_author(@book.author_id).where.not(id: @book.id)
+    return if @book
```

Comment:
check book có khác nil ko mới đi load recomment
 code `where not` em viết scope luôn